File: src/main/java/com/github/rooneyandshadows/lightbulb/annotation_processors/utils/TypeUtils.java
package com.github.rooneyandshadows.lightbulb.annotation_processors.utils;

import java.time.OffsetDateTime;
import java.util.Arrays;
import java.util.Date;
import java.util.List;

public class TypeUtils {
    public static final String stringType = String.class.getCanonicalName();
    public static final String intType = Integer.class.getCanonicalName();
    public static final String intPrimType = int.class.getCanonicalName();
    public static final String booleanType = Boolean.class.getCanonicalName();
    public static final String booleanPrimType = boolean.class.getCanonicalName();
    public static final String uuidType = java.util.UUID.class.getCanonicalName();
    public static final String floatType = Float.class.getCanonicalName();
    public static final String floatPrimType = float.class.getCanonicalName();
    public static final String longType = Long.class.getCanonicalName();
    public static final String longPrimType = long.class.getCanonicalName();
    public static final String doubleType = Double.class.getCanonicalName();
    public static final String doublePrimType = double.class.getCanonicalName();
    public static final String dateType = Date.class.getCanonicalName();
    public static final String OffsetDateType = OffsetDateTime.class.getCanonicalName();
    private static final List<String> simpleTypesList = Arrays.asList(stringType, intType, intPrimType, booleanType,
            booleanPrimType, uuidType, floatType, floatPrimType, longType, longPrimType, doubleType, doublePrimType);

    public static boolean isSimpleType(String canonicalName) {
        return simpleTypesList.contains(canonicalName);
    }
}

File: src/main/java/com/github/rooneyandshadows/lightbulb/annotation_processors/generator/utils/TypeUtils.kt
package com.github.rooneyandshadows.lightbulb.annotation_processors.generator.utils

import com.github.rooneyandshadows.lightbulb.annotation_processors.fragment.FragmentInfo
import com.squareup.kotlinpoet.ClassName
import java.time.OffsetDateTime
import java.util.*
import javax.lang.model.element.Element
import javax.lang.model.util.Elements

internal object TypeUtils {
    const val stringType: String = "java.lang.String"
    const val intType: String = "java.lang.Integer"
    const val intPrimType: String = "int"
    const val booleanType: String = "java.lang.Boolean"
    const val booleanPrimType: String = "boolean"
    const val uuidType: String = "java.util.UUID"
    const val floatType: String = "java.lang.Float"
    const val floatPrimType: String = "float"
    const val longType: String = "java.lang.Long"
    const val longPrimType: String = "long"
    const val doubleType: String = "java.lang.Double"
    const val doublePrimType: String = "double"
    const val dateType: String = "java.util.Date"
    const val OffsetDateType: String = "java.time.OffsetDateTime"

    private val simpleTypesList = listOf(
        String::class.qualifiedName, Int::class.qualifiedName, Boolean::class.qualifiedName, uuidType,
        Float::class.qualifiedName, Long::class.qualifiedName, Double::class.qualifiedName
    )

    fun isSimpleType(canonicalName: String): Boolean {
        return simpleTypesList.contains(canonicalName)
    }

    fun generateMappedFragmentBindingClassName(fragmentInfo: FragmentInfo, className: String): ClassName {
        return ClassName(fragmentInfo.className!!.packageName, className)
    }

    fun generateFragmentClassName(fragmentClass: Element, elements: Elements): ClassName {
        val classPackage: String = ElementUtils.getPackage(elements, fragmentClass)
        return ClassName(classPackage, fragmentClass.simpleName.toString())
    }
}